src/plugins/intel_gpu/src/graph/include/program_node.h

```diff
@@ -164,6 +166,10 @@ struct program_node {
     primitive_impl* get_selected_impl() const { return selected_impl.get(); }
     void set_selected_impl(std::unique_ptr<primitive_impl> impl);
 
+    void set_reuse_id(primitive_id id) { reuse_primitive_id = id; }
```

Comment:
The flag name is too general and maight be misread. 
(Because the memory reuse concept is already being applied w/ memorypool) 
For a special case, we need to come up with a name presenting the special purpose

Reply:
suggestion for naming
* `reuse_id()` -> `get_prim_id_for_output_buffer()`
* `set_reuse_id()` -> `set_prim_id_for_output_buffer()`
* `set_reuse_flag()` -> do we need this flag? I think we can check just with `prim_id_for_output_buffer`. If it is empty string, we can know that it is not used.